### DISCUSSION TITLE
Fixed and forced size when using the montage function ?

### DISCUSSION THREAD

**Discussion by echdareez** [ImageMagick/ImageMagick#7065] [Help] [answered]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 10 22H2

### Description

Hi there,

Just discovered ImageMagick and it seems to be perfect for the task I was hoping to do. Except for one tiny thing... 

Suppose I have a directory with 72 files and I want to join them together but with a fixed resolution (and in a grid of 10x7) : I've constructed this command : 
`**magick montage -size 6720x6552 -tile 10x7 -geometry +0+0 -border 0 *.jpg _sortthisoutlater_.jpg**`

The first 70 are properly placed in an image with the specified size, but the remaining 2 pictures are placed in a smaller image (as it's just 2 images on the 1st row). Is there any way to "force" the size? No matter what? And just fill up the rest of the image with the background color? 

thanks!
/e

UPDATE : There's a second step that can be executed (and does what I need) but I am still wondering if this can be done directly with the montage function : 
`**magick convert _sortthisoutlater_B.jpg -resize 6720x6552 -background white -gravity northwest -extent 6720x6552 test.jpg**`

**Comment by snibgo** [comment]
Your "-size 6720x6552" will have no effect.

>  ... but the remaining 2 pictures are placed in a smaller image (as it's just 2 images on the 1st row)

Is it smaller? How do you know? 

It will fill up the rest with the current background colour. Use `-background Red` or whatever you want. The default background is white.

**Reply by echdareez** [reply]
Thanks for the reply! 
It has indeed no effect, I've omitted it from the command and it worked the same :-) 

But yes, it is smaller - the images that I want to join, are all 672x936 - so the first picture is 6720x6552 (containing 70 pictures) while the second is 6720x936 (containing the remaining 2 pictures). My goal is to have this second picture to also be 6720x6552...

**Comment by snibgo** [comment, marked as answer]
Okay, it pads the first row, but not the entire image. I suggest you pipe the images to another `magick` command, which uses `-extent` to whatever size you need. In this example, I use red and blue colours for clarity.
```
magick montage mont_*.png -background Red -tile 2x1 -geometry +0+0 -border 0 miff:- |magick - -background Blue -extent 600x600 out.png
```

**Reply by echdareez** [reply]
Brilliant! That was it :-)   Thanks so much!
